app/deed/model.py

```diff
@@ -50,3 +79,9 @@ def matches(deed_id, borrower_id):
 
         for row in result:
             return int(row['count']) > 0
+
+    @staticmethod
+    def generate_token():
+        size = 6
+        return ''.join(random.choice(string.ascii_uppercase + string.digits)
```

Comment:
To ensure more likely uniqueness we could use:
`return str(uuid.uuid4().hex[:6]).lower()`

Since uuids put the highest entropy at the beginning it means the digits should have a high uniqueness.